source/host_resolver.c

```diff
@@ -1020,9 +1050,21 @@ static void resolver_thread_fn(void *arg) {
                 host_entry->host_name);
             aws_mutex_unlock(&host_entry->entry_lock);
 
-            AWS_ASSERT(err_code != AWS_ERROR_SUCCESS || aws_array_list_length(&callback_address_list) > 0);
+            size_t callback_list_size = aws_array_list_length(&callback_address_list);
+            if (callback_list_size > 0) {
+                AWS_LOGF_DEBUG(
```

Comment:
This should probably be TRACE?

Reply:
Honestly, I don't want any host resolution logging trace.  Given the periodic per-second nature of the default host resolver, finding issues gets real ugly with trace as we've already seen with the crt http client where the customer has to use dropbox/third party service just to give us logs that cover a few minutes of light usage.